src/librustdoc/visit_ast.rs

```diff
@@ -257,7 +228,9 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
         debug!("visiting item {:?}", item);
         let name = renamed.unwrap_or(item.ident.name);
 
-        if item.vis.node.is_pub() {
+        let is_pub = self.cx.tcx.visibility(item.def_id) == Visibility::Public;
```

Comment:
Hmm, I think this behaves differently for Inherited visibility: In `pub enum E { A, B }`, A and B will have `ty::Visibility::Public` if you call `visibility()`, but `ast::VisibilityKind::Inherited` if you call `node.is_pub()`. So this is a change in behavior, and I'm a little nervous about it.

That said, this does seem more correct to me - this should be using the computed visibility, not the syntactic visibility. Can you add a test that this now inlines enum variants? Right now it doesn't:
```rust
pub mod x {
    pub enum E { A, B }
}

#[doc(inline)]
pub use x::E::A;

#[doc(inline)]
pub use x::E;
```
![image](https://user-images.githubusercontent.com/23638587/129508496-8d5b7594-fed9-4410-9158-85625badda79.png)

Reply:
Actually, I'm not sure what it means to inline variants :/ they don't get their own documentation page (and the same for trait methods, which also have inherited visibility). So please do add the test, I'm curious to see what it changes. We may need to keep the original behavior depending on what it does.

Reply:
Tested locally. It doesn't seem to do anything. The inline attribute is silently ignored, the same way it is on stable. I'll add an actual test for it when I get a chance.

Reply:
I think it might be better for me to separate the change over to `tcx.visibility` into a separate PR and do something more conservative here. It seems like this might cause a regression and we might end up having to revert. Thoughts?

Reply:
A separate PR is possible, yes.